src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/NotSuitableSDOCallbackHandler.java

```diff
@@ -94,15 +95,24 @@ private CallbackResponse validateNotSuitableReason(CallbackParams callbackParams
         final int lengthAllowed = 150;
         List<String> errors = new ArrayList<>();
         String reason;
-        if (isTransferOnlineCase(callbackParams.getCaseData())) {
-            reason = callbackParams.getCaseData().getTocTransferCaseReason().getReasonForCaseTransferJudgeTxt();
+
+        if (toggleService.isSdoR2Enabled()
+            && callbackParams.getCaseData().getNotSuitableSdoOptions() == NotSuitableSdoOptions.CHANGE_LOCATION
+            && callbackParams.getCaseData().getCcdState() == CaseState.CASE_PROGRESSION) {
+            errors.add(
+                "Unable to process this request. To transfer the case to another court you need to issue a General Order.");
         } else {
-            reason = callbackParams.getCaseData().getReasonNotSuitableSDO().getInput();
-        }
-        if (reason.length() > lengthAllowed) {
-            errors.add("Character Limit Reached: "
-                + "Reason for not drawing Standard Directions order cannot exceed "
-                + lengthAllowed + " characters.");
+            if (isTransferOnlineCase(callbackParams.getCaseData())) {
```

Comment:
Review this code. I think logic related to both flags "toggleService.isSdoR2Enabled()", "isTransferOnlineCase(callbackParams.getCaseData()" should be independent.